packages/nx/src/lock-file/utils/mapping.ts

```diff
@@ -67,21 +67,28 @@ function mapTransitiveDependencies(
     if (versionCache[key]) {
       result.push(versionCache[key]);
     } else {
-      const matchedVersion = versions[`${packageName}@${version}`]
-        ? version
-        : transitiveLookupFn({
-            packageName,
-            parentPackages,
-            versions,
-            version,
-          })?.version;
+      let matchedVersion: string;
+      let isRootVersion: boolean;
+      if (versions[`${packageName}@${version}`]) {
+        matchedVersion = version;
+        isRootVersion = versions[`${packageName}@${version}`].rootVersion;
+      } else {
+        const transitiveLookupResult = transitiveLookupFn({
+          packageName,
+          parentPackages,
+          versions,
+          version,
+        });
+        matchedVersion = transitiveLookupResult?.version;
+        isRootVersion = transitiveLookupResult?.rootVersion;
+      }
 
       // for some peer dependencies, we won't find installed version so we'll just ignore these
       if (matchedVersion) {
         const nodeName = getNodeName(
           packageName,
           matchedVersion,
-          versions[`${packageName}@${matchedVersion}`]?.rootVersion
+          isRootVersion
```

Comment:
So this is the core of the issue?

Currently, the rootVersion flag here belonged to the root version and not the transitive lookup's result root version?

Reply:
Yes, the problem was that for alias packages the matchedVersion was correct but the key for versions was not `${packageName}@${matchedVersion}` we always for `undefined` there as `rootVersion`